etcdserver/server.go

```diff
@@ -969,6 +969,9 @@ func (s *EtcdServer) Stop() {
 		plog.Warningf("%s failed to transfer leadership (%v)", s.ID(), err)
 	}
 	s.HardStop()
+	if s.authStore != nil {
```

Comment:
this teardown code should be in the big defer in `EtcdServer.run()` before the backend closes